UnityPy/files/BundleFile.py

```diff
@@ -301,31 +301,23 @@ def save_fs(self, writer: EndianBinaryWriter, data_flag: int, block_info_flag: i
 
         file_data = data_writer.bytes
         data_writer.dispose()
-        uncompressed_data_size = len(file_data)
 
-        # compress the data
-        switch = block_info_flag & 0x3F
-        if switch == 1:  # LZMA
-            file_data = CompressionHelper.compress_lzma(file_data)
-        elif switch in [2, 3]:  # LZ4, LZ4HC
-            file_data = CompressionHelper.compress_lz4(file_data)
-        elif switch == 4:  # LZHAM
-            raise NotImplementedError
-        # else no compression - data stays the same
-        compressed_data_size = len(file_data)
+        file_data, block_info = CompressionHelper.chunk_based_compress(
+            file_data, block_info_flag
+        )
 
         # write the block_info
         # uncompressedDataHash
         block_writer = EndianBinaryWriter(b"\x00" * 0x10)
         # data block info
-        # block count
-        block_writer.write_int(1)
-        # uncompressed size
-        block_writer.write_u_int(uncompressed_data_size)
-        # compressed size
-        block_writer.write_u_int(compressed_data_size)
-        # flag
-        block_writer.write_u_short(block_info_flag)
+        block_writer.write_int(len(block_info))
+        for u_size, c_size, f in block_info:
```

Comment:
Can you please use the complete meaning as name,
so block_uncompressed_size, block_compressed_size, block_flag.

Reply:
Thanks.